### PR TITLE
[patch] Add enablelayer to grammar.

### PR DESCRIPTION
**Pull request by dtzSiFive**

### CODE REVIEW

**Comment by dtzSiFive** [comment]
Any suggestions for org welcome!

File: spec.md [hunk 1]
```diff
@@ -4358,6 +4359,9 @@ property_primop_2expr = property_primop_2expr_keyword ,
 (* Tokens: Annotations *)
 annotations = "%" , "[" , json_array , "]" ;
 
+(* Enable Layer *)
+enablelayer = "enablelayer" , id , { "." , id };
```
Comment (mmaloney-sf) [inline spec.md:4363]: Could we introduce a new production rule for colorings?

`coloring := id , { "." , id } ;`

Also, to be consistent with the other rules, let's put a space before the semicolon.

File: spec.md [hunk 2]
```diff
@@ -4358,6 +4359,9 @@ property_primop_2expr = property_primop_2expr_keyword ,
...
+
```
Comment (seldridge) [inline spec.md:4364]: If this is inlined above, does that fit on one line still? (Or, can it be made to split with line splitting?)

File: spec.md (hunk 1 above)
Comment (dtzSiFive) [reply spec.md:4363]: Oops dropped the space when reworking, good catch ty!

File: spec.md (hunk 2 above)
Comment (dtzSiFive) [reply spec.md:4364]: I'll check.  I broke it out to not have to wrap the line from what it had already, but since I did I'll give that a go :+1:.

Comment (mmaloney-sf) [reply spec.md:4364]: Inlining it would lead to doubly-nested `{` `}`'s. 

It might be good to leave this split out for that reason.

**Review by seldridge — APPROVED** [review]
LGTM

File: spec.md (hunk 2 above)
Comment (mmaloney-sf) [reply spec.md]: More practically, some parser generators would not accept the inlined version because of the doubly-nested `{` `}`.

Comment (seldridge) [reply spec.md]: I'm fine with splitting it out. @mmaloney-sf: Do you know where this should live? It was previously under the "token" section.  Is that appropriate or could you propose an alternative?

Comment (mmaloney-sf) [reply spec.md]: For nonterminals which are only used once, it's probably good to keep them near their usage point.

But it's easier for us to shuffle definitions than to refactor the grammar, so anywhere is fine by me.

Comment (dtzSiFive) [reply spec.md]: Great, wasn't sure how to balance against the idea to sink less-important things later (e.g., `annotations`).

But yes easy to rearrange as we see fit at any point.

Thanks for your input all!

**Pull request merged**